### ISSUE TITLE
[Feature Request] : Support GFM Alerts / Notes 

### ISSUE THREAD

**Issue by calumk** [mdit-plugins/mdit-plugins#33]
### Checklist

- [X] I searched existing issues and no one else requests similar feature.
- [X] I think that 25%+ users are positive towards this feature.

### Describe the feature

Support GFM Alerts / Notes 

GFM allow for markup such as :

> [!Important]
> This is an important note

> [!Warning]
> This is a warning

```
> [!Important]
> This is an important note

> [!Warning]
> This is a warning
```

https://github.com/orgs/community/discussions/16925

It is not the same format used by the containers plugin : https://mdit-plugins.github.io/container.html

### Additional context

Reference - It should be a plugin : https://github.com/markdown-it/markdown-it/issues/974#issuecomment-1819253627

_No response_

**Comment by Mister-Hope** [comment]
I'm really busy these days writing my article,  I'm a post graduate student.

PR welcome

**Comment by Mister-Hope** [comment]
> [!Warning]
> This is a warning


> [!Warning]
>
>
> This is a warning


> [!Warning]:
>
>
> This is a warning


> [!warning]
>
>
> This is a warning


> [!waRNing]
>
> This is a warning


> [!Warning]
> This is a warning
> > [!Note]
> > This is a note

```md

> [!Warning]
> This is a warning


> [!Warning]
>
>
> This is a warning


> [!Warning]:
>
>
> This is a warning


> [!warning]
>
>
> This is a warning


> [!waRNing]
>
> This is a warning


> [!Warning]
> This is a warning
> > [!Note]
> > This is a note
```

**Comment by Mister-Hope** [comment]
It seems that this is not case sensitive, but do not support nesting.

**Comment by Mister-Hope** [comment]
Space test:

> [!Warning]
> This is a warning

> [!Warning] 
> This is a warning

>  [!Warning]
> This is a warning

>   [!Warning]
> This is a warning

>    [!Warning]          
> This is a warning

```md
> [!Warning] 
> This is a warning

> [!Warning]  
> This is a warning

>  [!Warning]
> This is a warning

>   [!Warning]
> This is a warning


>    [!Warning]          
> This is a warning
```

**Comment by Mister-Hope** [comment]
> [Warning] 
> This is a warning

> ![Warning] 
> This is a warning

**Comment by Mister-Hope** [comment]
Cannot more than 4 space at beginning:

>     [!Warning]
> This is a warning

**Comment by Mister-Hope** [comment]
No nesting support and only at first level:

> [!Warning]
> This is a warning
> > [!Note]
> > This is a note

- > [!Warning]
  > This is a warning

**Comment by calumk** [comment]
This is not working as expected - it seem to work, but not if you use it in a real-world use case by addding a paragraph before the alert - 

```
## Title Example

> [!note]
> Warning Text

> [!tip]
> Tip Text
```

```
<h2>Title Example</h2>
<div class="markdown-alert markdown-alert-note">
<p class="markdown-alert-title">Note</p>
<p>Warning Text</p>
</div>
<div class="markdown-alert markdown-alert-tip">
<p class="markdown-alert-title">Tip</p>
<p>Tip Text</p>
</div>
```


but add a paragraph before and the output does not work

```
## Title Example

This is a paragraph

> [!note]
> Warning Text

> [!tip]
> Tip Text

```
<h2>Title Example</h2>
<p>This is a paragraph</p>
<blockquote>
<p>[!note]
Warning Text</p>
</blockquote>
<blockquote>
<p>[!tip]
Tip Text</p>
```

**Comment by calumk** [comment]
Seems to be related to paragraphs, it works after some elements like `---` but not `paragrahs` or `images` or `links`

**Comment by calumk** [comment]
Seems to fail if there is a paragraph anywhere above in the document 


## FAIL : 
```
## Title Example

Example Paragraph

---

> [!note]
> Warning Text

> [!tip]
> Tip Text

```


## PASS : 

```
## Title Example

---

> [!note]
> Warning Text

> [!tip]
> Tip Text

```

**Comment by Mister-Hope** [comment]
That's werid

**Comment by Mister-Hope** [comment]
fixed in 0.7.1
